Source/BugsnagSessionTracker.m

```diff
@@ -138,6 +138,10 @@ - (void)startNewSessionIfAutoCaptureEnabled {
 }
 
 - (void)startNewSessionWithAutoCaptureValue:(BOOL)isAutoCaptured {
+    NSArray *releaseStages = self.config.enabledReleaseStages;
+    if (releaseStages != nil && ![releaseStages containsObject:self.config.releaseStage]) {
+        return;
+    }
```

Comment:
I'm inclined to say that any code changes should be on a separate PR, as we could easily miss an explanation for changed behaviour (when the title just says "Add tests").  I'd also want to make sure it receives sufficient attention from a Cocoa expert.  E.g. are there cases we still need to consider here (`releaseStages` being `nil`, perhaps)?